src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,89 @@
-import React from 'react';
+import React, { useState } from 'react';
+import cn from 'classnames';
 import './FindMovie.scss';
+import { MovieCard } from '../MovieCard';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+
+type Props = {
+  onClickAddMovie: (movie: Movie) => void,
+};
+
+const DEFAUT_POSTER_IMG = 'https://via.placeholder.com/'
++ '360x270.png?text=no%20preview';
+const MOVIE_IMDB_URL = 'https://www.imdb.com/title/';
+
+export const FindMovie: React.FC<Props> = ({ onClickAddMovie }) => {
+  const [loadedMovie, setLoadedMovie] = useState<Movie | null>(null);
+  const [hasSuccessResponse, setHasSuccessResponse] = useState(true);
+  const [query, setQuery] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleFindMovie = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    try {
+      setIsLoading(true);
+      setHasSuccessResponse(true);
+      const movieData = await getMovie(query);
+
+      if ('Error' in movieData) {
+        throw new Error(`Can't find movie such a ${query}`);
+      }
+
+      const {
+        Title,
+        Poster,
+        Plot,
+        imdbID,
+      } = movieData;
+
+      const newMovie: Movie = {
+        title: Title,
+        description: Plot,
+        imgUrl: Poster === 'N/A'
+          ? DEFAUT_POSTER_IMG
+          : Poster,
+        imdbUrl: `${MOVIE_IMDB_URL}${imdbID}`,
+        imdbId: imdbID,
+      };
+
+      setLoadedMovie(newMovie);
+    } catch (error) {
+      setHasSuccessResponse(false);
+    } finally {
+      setIsLoading(false);
+    }
+  };
```

Comment:
Here i here I suppose you can handle it without - try-catch. It's enough to check the response like you did in if, just add else for another response. but it's only my opinion

Reply:
Agree, but I thought it would be good to remain it, cause of the situation if the function will return Promise with reject, then this catch method will save us and everything will be steelworking as it should be.